packages/contracts/contracts/LenderCommitmentForwarder/extensions/LenderCommitmentGroup/LenderCommitmentGroup_Smart.sol

```diff
@@ -390,6 +397,18 @@ contract LenderCommitmentGroup_Smart is
         );
     }
 
+    function prepareSharesForWithdraw(
+        uint256 _amountPoolSharesTokens 
+    ) external returns (bool) {
+        require( poolSharesToken.balanceOf(msg.sender) >= _amountPoolSharesTokens  );
+
+        poolSharesPreparedToWithdrawForLender[msg.sender] = _amountPoolSharesTokens; 
+        poolSharesPreparedTimestamp[msg.sender] = block.timestamp;
```

Comment:
make these variables be  reset when taking certain actions like lender deposit principal